src/store.c

```diff
@@ -0,0 +1,245 @@
+/*
+ *   Copyright (C) 2023 SUSE LLC
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, write to the Free Software
+ *   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ *
+ * Written by Olaf Kirch <okir@suse.com>
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include "store.h"
+#include "util.h"
+#include "tpm.h"
+
+/* We do not have automatic conversion of TPM2B_PUBLIC to RSA yet (and so far we haven't needed it) */
+#undef WITH_NATIVE_TO_RSA_CONVERSTION
+
+tpm_rsa_key_t *
+stored_key_read_rsa_private(const stored_key_t *sk)
+{
+	switch (sk->format) {
+	case STORED_KEY_FMT_PEM:
```

Comment:
why not a simple `if`? are there more conditions?

Should the `stored_key_t->is_private` validated?

Reply:
There probably will be a STORED_KEY_FMT_NATIVE case.